crates/simulate/src/manager.rs

```diff
@@ -44,99 +53,115 @@ impl<U: UpdateData, W: World<WorldUpdateData = U>> SimManager<U, W> {
+                agent_channel_receivers.insert(*addr, channel_receiver);
+            }
 
-            // Concurrently:
-            //      1) Spawn agents to access immutable state via a read handle and queue updates.
-            //      2) Append queued updates via the write handle.
             rayon::scope(|s| {
-                for (agent_id, agent) in &mut self.agents {
-                    let channel: AgentChannel<U> = AgentChannel::new(&sender, agent_id);
-                    s.spawn(|_| agent.step(&self.state, channel));
+                for (agent_id, agent) in self.agents.iter_mut() {
+                    s.spawn(|_| {
+                        agent.step(
+                            &self.state,
+                            agent_channel_senders
+                                .get(agent_id)
+                                .expect("Agent id must be in map."),
+                        );
+                    });
                 }
-                drop(sender);
             });
 
-            for update in receiver.iter() {
+            while let Ok(update) = evm_update_receiver.try_recv() {
+                self.state
+                    .execute_evm_tx_state_update(update)
+                    .map_err(SimManagerError::StateError)?;
+            }
+
+            while let Ok(update) = world_update_receiver.try_recv() {
                 self.state
-                    .execute(update)
-                    .map_err(SimManagerError::SimStateError)?;
+                    .execute_world_state_update(update)
+                    .map_err(SimManagerError::StateError)?;
             }
 
-            // Let agents resolve the step.
-            thread::scope(|t| {
-                for agent in self.agents.values_mut() {
-                    t.spawn(|| agent.resolve_step(&self.state));
+            rayon::scope(|s| {
+                for (addr, agent) in self.agents.iter_mut() {
+                    s.spawn(|_| {
+                        agent.resolve_step(
+                            &self.state,
+                            agent_channel_receivers
+                                .get(addr)
+                                .expect("Agent id must be in map."),
+                        );
+                    })
                 }
             });
 
             // Run summarizers.
             self.summarizer.output_summaries(&self.state);
 
-            // Clear all results.
-            self.state.clear_all_results();
-
             // Update time policy.
-            self.state.update_time_env(self.time_policy.step());
+            self.state.update_time(self.time_policy.step());
         }
 
         Ok(())
     }
 
     /// Adds and activates an agent to be put in the collection of agents under the manager's control.
-    /// # Arguments
-    /// * `new_agent` - The agent to be added to the collection of agents.
     pub fn activate_agent(&mut self, mut new_agent: Box<dyn Agent<U, W>>) -> SimManagerResult<()> {
```

Comment:
Activate agent is much simpler now since agents are passed a mutable ref to state.